src/transforms/groupby.js

```diff
@@ -137,8 +137,22 @@ function pasteArray(newTrace, trace, j, a) {
     );
 }
 
+// In order for groups to apply correctly to other transform data (e.g.
+// a filter transform), we have to break the connection and clone the
+// transforms so that each group writes grouped values into a different
+// destination. This function does not break the array reference
+// connection between the split transforms it creates. That's handled in
+// initialize, which creates a new empty array for each arrayAttr.
+function cloneTransforms(newTrace) {
```

Comment:
Oh hah - of course, because the original `Lib.extendDeepNoArrays({}, trace)` doesn't dive into the `transforms` array. Do we need a `Lib.extendDeepOnlyTheseArrays({}, trace, ['transforms'])` ? 🙈 

> is that something we should do for all transforms?

This only applies to transforms that generate multiple traces, right? I do kind of like the idea of recursive transforms, but until we do something like that, that might be more inherently robust, I think we should manage this case-by-case.